packages/web-components/fast-element/src/templating/when.ts

```diff
@@ -1,24 +1,38 @@
-import type { Binding, ExecutionContext } from "../observation/observable.js";
+import { isFunction } from "../interfaces.js";
+import type { ExecutionContext, Expression } from "../observation/observable.js";
 import type { CaptureType, SyntheticViewTemplate } from "./template.js";
 
+const noTemplate: () => null = () => null;
+function normalizeBinding<TSource>(
+    value: SyntheticViewTemplate | Expression<TSource, SyntheticViewTemplate> | undefined
+): (() => null) | Expression<TSource, SyntheticViewTemplate> {
+    return value === undefined ? noTemplate : isFunction(value) ? value : () => value;
+}
+
 /**
  * A directive that enables basic conditional rendering in a template.
- * @param binding - The condition to test for rendering.
+ * @param condition - The condition to test for rendering.
  * @param templateOrTemplateBinding - The template or a binding that gets
  * the template to render when the condition is true.
+ * @param elseTemplateOrTemplateBinding - Optional template or binding that that
+ * gets the template to render when the conditional is false.
  * @public
  */
-export function when<TSource = any, TReturn = any>(
-    binding: Binding<TSource, TReturn>,
+export function when<TSource = any, TReturn = any, TParent = any>(
+    condition: Expression<TSource, TReturn, TParent> | boolean,
```

Comment:
I think we might want to keep this signature the same instead of adding a new `Expression` type since that type is new in FE2. Probably best to just add a third argument of `SyntheticViewTemplate | Binding<TSource, SyntheticViewTemplate>`